File: src/client/java/com/somefrills/features/core/Feature.java
package com.somefrills.features.core;

import io.github.notenoughupdates.moulconfig.observer.Property;

public abstract class Feature extends AbstractFeature {

    public Feature(Property<Boolean> enabledProperty) {
        super(enabledProperty);
    }

    @Override
    protected final boolean evaluate() {
        return true;
    }

    @Override
    protected final void onEnable() {
    }

    @Override
    protected final void onDisable() {
    }
}

File: src/main/kotlin/com/somefrills/features/core/Feature.kt
package com.somefrills.features.core

import io.github.notenoughupdates.moulconfig.observer.Property

abstract class Feature(enabledProperty: Property<Boolean>) : AbstractFeature(enabledProperty) {
    override fun evaluate(): Boolean {
        return true
    }

    override fun onEnable() {
    }

    override fun onDisable() {
    }
}